lib/ansible/modules/cloud/amazon/ec2.py

```diff
@@ -1227,12 +1227,15 @@ def create_instances(module, ec2, vpc, override_count=None):
                 # Now we have to do the intermediate waiting
                 if wait:
                     instids = await_spot_requests(module, ec2, res, count)
+                else:
+                    instids = []
```

Comment:
Do we also need to guard the code here: https://github.com/ansible/ansible/pull/23053/files#diff-dea76a894873bd8fbd3173b3a8610a11L1280 or does  ec2.create_tags() handle an empty list fine?